### DISCUSSION TITLE
Changing Express to Nest?

### DISCUSSION THREAD

**Discussion by kol3x** [TheOdinProject/curriculum#27758] [Ideas]
With Nest.js passing Express in GitHub stars a few weeks ago I wonder if TOP should teach it instead.

I rarely see Express as a requirement in job postings, but I see a lot of Nest. 

Personally, I got into Nest recently and I feel like it gives a much better experience for building backend, giving you consistent structure that you can follow. From the first glance, it looks more complex, but in the end it only makes the development easier.

What do you think?

**Comment by CouchofTomato** [comment]
@TheOdinProject/javascript Any views on this please?

**Comment by mao-sz** [comment]
Thank you for opening this discussion, @kol3x.

On the contrary to your experience, I have barely seen NestJS jobs in my country (inc. remote) but plenty more that mention Express. This will likely differ from country to country.

Ultimately, what we pick for the curriculum isn't driven solely by what's most popular in the market or not, but also driven by how the tool can help us teach the concepts we feel should be taught. I.e. it should fit the pedagogy first and foremost.

The JS ecosystem is really diverse and there are new frameworks and libraries being spawned by the minute, with everyone wanting to do things their way. Ultimately, the JS ecosystem as a whole is pretty unopinionated. I think it makes more sense pedagogically to keep using Express in our curriculum, as it's more than sufficient for helping us teach the concepts we wish to teach, the way we wish to have them taught. NestJS AFAIK is built on top of Express by default anyway.

And if there's an argument to switch to NestJS, then why not also any other JS backend framework? Because I'm sure different people can certainly come up with individual arguments for why they feel Fastify might be better for the curriculum, or Sails or whatever.

IMO, Express is perfectly fine for our goals, so switching to NestJS wouldn't be solving any significant problems that I can see. Perhaps @01zulfi might have more to share on this, as he's the main one organising the Node Revamp.

**Comment by mao-sz** [comment]
Going to close this as we currently do not plan to change the Node course to use Nest instead of Express. IMHO, in the JS ecosystem, it's probably better pedagogically to teach these concepts via Express to empower learners to learn what they need for other more opinionated frameworks when necessary.
